python/xorbits/_mars/services/task/task_info_collector.py

```diff
@@ -139,46 +150,50 @@ async def collect_subtask_operand_structure(
                         ):
                             chunk_dict["dtype"] = str(input_chunk.dtype)
 
-                        op_dict["inputs"][input_chunk.key] = chunk_dict
+                        op_info["inputs"][input_chunk.key] = chunk_dict
 
-                op_save_path = os.path.join(
-                    self._save_dir, f"{stage_id[:trunc_key]}_operand.yaml"
-                )
-                await self._save_task_info({op.key: op_dict}, op_save_path)
+                op_info[op.key] = op_info
                 visited.add(op.key)
 
-            subtask_save_path = os.path.join(
-                self._save_dir, f"{stage_id[:trunc_key]}_subtask.yaml"
-            )
-            await self._save_task_info(
-                {subtask.subtask_id: subtask_dict}, subtask_save_path
-            )
+            subtask_info[subtask.subtask_id] = subtask_info
+
+        asyncio.create_task(self._save_task_info(op_info, op_save_path))
+        asyncio.create_task(self._save_task_info(subtask_info, subtask_save_path))
```

Comment:
As all collections call `_save_task_info`, just modify the `_save_task_info` function.
`await task_api.save_task_info(task_info, path)` ->
`asyncio.create_task(task_api.save_task_info(task_info, path))`